src/main/c/cext/printf_musl.c

```diff
@@ -0,0 +1,117 @@
+#include <truffleruby-impl.h>
+#include <ffi.h>
```

Comment:
This needs below the `#if`, otherwise it fails to compile on other platforms.
I'd make the `#if defined(__linux__) && !defined(__GLIBC__)` the first line actually.

The file should also have a header like sibling files:
```c
/*
 * Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved. This
 * code is released under a tri EPL/GPL/LGPL license. You can use it,
 * redistribute it and/or modify it under the terms of the:
 *
 * Eclipse Public License version 2.0, or
 * GNU General Public License version 2, or
 * GNU Lesser General Public License version 2.1.
 */
```

Reply:
Oops, thanks for catching this. But we need to include system headers first, as they define the symbols we're checking